packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -212,14 +212,9 @@ export class RestApi extends Resource implements IRestApi {
    */
   public deploymentStage!: Stage;
 
-  /**
-   * The domain name mapped to this API, if defined through the `domainName`
-   * configuration prop.
-   */
-  public readonly domainName?: DomainName;
-
   private readonly methods = new Array<Method>();
   private _latestDeployment: Deployment | undefined;
+  private _domainNames: DomainName[] = [];
```

Comment:
Curious to know why you've chosen the approach to store all of `DomainName` even though they're all not needed.

You could instead have stored the first one with a conditional on `addDomainName()`.

Is there some value you see in storing all of them here? I'm tilting towards only storing what's necessary so the amount of state stored is minimal.

Reply:
probably a premature code from my side. but, I assumed that CDK users might want to be able to iterate through all domains associated with an API Gateway, to be able to change things like Security Policy, Endpoint Configuration, & the Certificate associated with the domain. So, this was an attempt to future proof for that possibility.

Reply:
I'd rather not. Let's make that change when/if it comes. We may end up deciding that there is a better way to do that. 

Let's change this to do only what's necessary here.